src/hooks/useV2Pairs.ts

```diff
@@ -22,82 +23,91 @@ export enum PairState {
 }
 
 export function useV2Pairs(currencies: [Currency | undefined, Currency | undefined][]): [PairState, Pair | null][] {
-  const [reserves, setReserves] = useState<
-    ({ reserve0: string; reserve1: string; poolId: number; totalSupply: string } | number | undefined)[]
-  >([]);
-
   const tokens = useMemo(() => {
     return currencies.map(([currencyA, currencyB]) => [currencyA?.wrapped, currencyB?.wrapped]);
   }, [currencies]);
 
-  // const lastBlockNumber = useBlockNumber();
+  const [pairs, setPairs] = useState<[PairState, Pair | null][]>(Array(tokens.length).fill([PairState.LOADING, null]));
+
+  const lastBlockNumber = useBlockNumber();
 
   useEffect(() => {
-    setReserves(Array(tokens.length).fill(PairState.LOADING));
+    setPairs(Array(tokens.length).fill([PairState.LOADING, null]));
```

Comment:
Any reason to extract it as a separate hook?

Reply:
Could you explain more?

Reply:
The reason is to keep previous data. This is why I refactored the code. @lijac